collections/_posts/2023-07-10-self-explaining-neural-networks.md

```diff
@@ -0,0 +1,127 @@
+1. **Explicitness/Intelligibility**: _are the explanations immediately understandable?_
+2. **Faithfulness**: _are relevance scores indicative of "true" importance?_
+3. **Stability**: _how consistent are explanations for similar/neighboring examples?_
+
+## Datasets used
+
+- MNIST & CIFAR10
+  - Images classification
+  - 32x32 & 32x32x3
+- Several datasets from UCI Machine Learning Repository
+  - Regression, classification, segmentation tasks
+  - Data types are numerical, categories, images, ...
+- Propublica's COMPAS Recidivism Risk Score
+  - COMPAS is a closed-source model used by USA DoJ to evaluate probability of recidivism
+  - Dataset consists of results from the score, often used to evaluate fairness of predictive algorithms
+
+## Architectures details and accuracy results
+
+The authors do not detail their predictive performances on previously mentioned datasets, only that the proposed models are
+> on-par with their non-modular, non interpretable counterparts.
+
+They just precise achieving less than 1.3% error rate on MNIST test set or an accuracy of 78.56% on CIFAR10 test set, which is "on par for models of that size trained with some regularization method".
+
+For each task, they used different architectures for the sub-branches $$h (\cdot)$$ and $$\theta (\cdot)$$ of the SENN model. FC stands for _Fully connected_ layer, and CL for _Convolutional_ layer. The multiplicative constant 10 in the final layer of the $$\theta$$ function for both MNIST and CIFAR10 corresponds to the number of classes. In all cases, the training occurred using Adam optimizer with a learning rate $$l = 2 \times 10^{-4}$$, and the sparsity of learned $$h(\cdot)$$ was enforced through $$\xi = 2 \times 10^{-5}$$.
+
+![](/collections/images/SENN-model/architectures-used-per-dataset.jpg)
+
+## Intelligibility
+
+Explanations of this model are to be understood relative to the concepts defined. Here, they propose to retrieve examples of the dataset which maximally activate specific concepts.
+
+![](/collections/images/SENN-model/figure2.jpg)
+
+## Stability
+
+Empiric tests show low robustness of interpretability methods to local perturbations of the input. Small changes in the input produce visible modifications of the explanations
+
+![](/collections/images/SENN-model/figure4.jpg)
+
+The Self-Explaining Neural Network model is less subject to adverse effects of input perturbation on explanations stability.
+
+![](/collections/images/SENN-model/figure8.jpg)
+
+# Limitations
+
+- No details on model performances, even in supplementary materials
+- Definition of interpretable concepts is a problem in itself
+- The final prediction is more explainable, but the intermediate steps are not: the concept encoder and relevance parametrizer are still complete black-boxes
+
+# References
+
+[^1]: [Alvarez-Melis D. and Jaakkola T.S., On the Robustness of Interpretability Methods. ]()
```

Comment:
The reference here seems incomplete and the link itself is missing.

Reply:
Woups, I've done it too quickly at the last minute. I'll correct it!